source/jormungandr/jormungandr/parking_space_availability/car/common_car_park_provider.py

```diff
@@ -73,9 +83,32 @@ def get_informations(self, poi):
         if data:
             return self.process_data(data, poi)
 
-    def support_poi(self, poi):
+    def _is_poi_coords_within_shape(self, poi):
+        if self.shape is None:
+            return True
+
+        try:
+            coord = poi['coord']
+            coords = [float(coord['lon']), float(coord['lat'])]
+            return self.shape.contains(shapely.geometry.Point(coords))
+        except KeyError as e:
+            self.log.error(
+                "Coords illformed, 'poi' needs a coords dict with 'lon' and 'lat' attributes' : {}".format(
+                    str(e)
+                )
+            )
+        except Exception as e:
+            self.log.error("Cannot find if coords is within shape: {}".format(str(e)))
+
+        return False
+
+    def _has_supported_poi_operator(self, poi):
         properties = poi.get('properties', {})
-        return properties.get('operator', '').lower() in self.operators
+        operator = properties.get('operator', '').lower()
+        return operator in self.operators
```

Comment:
Why separate it in 2 lines?

Reply:
I did that to ease readability for the same as for the above line. Otherwise, I would have ended-up with something like:

```python
        return poi.get('properties', {}).get('operator', '').lower() in self.operators
```